homeassistant/components/tado/__init__.py

```diff
@@ -169,6 +170,33 @@ def __init__(self, hass, username, password, fallback):
             "zone": {},
         }
 
+    def decide_overlay_mode(self, duration, zone_id, overlay_mode=None):
+        """Return correct overlay mode based on the action and defaults."""
+        # If user gave duration then overlay mode needs to be timer
+        if duration:
+            overlay_mode = CONST_OVERLAY_TIMER
+        # If no duration or timer set to fallback setting
+        if overlay_mode is None:
+            overlay_mode = (
+                self.fallback if self.fallback is not None else CONST_OVERLAY_TADO_MODE
+            )
+        zone_data = self.data["zone"][zone_id]
+        # If default is Tado default then look it up
+        if overlay_mode == CONST_OVERLAY_TADO_DEFAULT:
+            overlay_mode = (
+                zone_data.default_overlay_termination_type
+                if zone_data.default_overlay_termination_type is not None
+                else CONST_OVERLAY_TADO_MODE
+            )
+        # If we ended up with a timer but no duration, set a default duration
+        if overlay_mode == CONST_OVERLAY_TIMER and duration is None:
+            duration = (
+                zone_data.default_overlay_termination_duration
+                if zone_data.default_overlay_termination_duration is not None
+                else "3600"
+            )
```

Comment:
This statement has no effect and, in fact, should not be extracted. In the files below, `overlay_mode: str: None` indeed depends on `duration: str | None`; but later `duration` is _mutated_ via assigning default value to it, and passed to `self._tado.set_zone_overlay` in both cases.